File: Arrochado.java
package com.company;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;
import java.util.Scanner;

public class Arrochado {

    public static void main(String[] args) {
	    arrochado();
    }

    private static void arrochado() {

        Scanner keyboard = new Scanner(System.in);

        int random = getRandomNumberUsingNextInt();

        List<Integer> tries = new ArrayList<Integer>();
        int maior = 0;
        int menor = 0;
        int amount = 0;
        int extremidade_menor = 0;
        int extremidade_maior = 100;
        int input = 0;
        int aux = 0;

        // estatisticas
        float media = 0F;

        System.out.println("Digite seu nome: ");
        String name = keyboard.nextLine();

        do {

            System.out.println("Digite seu palpite!\nDICA: O número está no intervalo de " + extremidade_menor + " ~ " + extremidade_maior + ":");
            input = keyboard.nextInt();

            if (input > extremidade_maior || input < extremidade_menor) {
                System.out.println("=>O numero digitado nao se encontra no range determinado de " + extremidade_menor + " ~ " + extremidade_maior + ". Tente novamente.");
            } else {

                if(amount == 0)
                    menor = input;

                // armazena o maior e menor valor
                if (input > maior)
                    maior = input;
                else  if (input < menor)
                    menor = input;

                if (input == random){
                    tries.add(input);
                    amount++;
                    break;
                }

                if (random > input)
                    extremidade_menor = input;
                else if (random < input)
                    extremidade_maior = input;

                tries.add(input);
                amount++;

                System.out.println("=>Valor errado, tente novamente!");
            }
        }while (input != random);

        System.out.println("=> Parabéns " + name + ",\nVocê acertou o número!");

        tries = ordenar(tries, aux);
        media = media(tries);

        System.out.println("\n==> O maior numero digitado foi " + maior + " e o menor " + menor);
        System.out.printf("\n==> A media dos numeros digitados foi: %.2f", media);

        estatisticas(tries, media);
    }

    private static int getRandomNumberUsingNextInt() {
        Random random = new Random();
        return random.nextInt(100 + 1);
    }

    private static List<Integer> ordenar (List<Integer> list, int aux) {

        for (int i = 0; i < list.size(); i++) {
            for (int j = 0; j < list.size(); j++) {
                if (list.get(i) < list.get(j)) {
                    aux = list.get(i);
                    list.set(i, list.get(j));
                    list.set(j, aux);
                }
            }
        }

        System.out.printf("\n==> Seus números: ");
        for (int num : list) {
            System.out.printf("[" + num + "]");
        }

        return list;
    }

    private static float media (List<Integer> list) {

        float somatorio = 0;
        for (int num : list) {
            somatorio += num;
        }
        return somatorio /= list.size();
    }

    private static void estatisticas(List<Integer> list, float media) {

        float mediana = 0F;
        float variancia = 0F;
        float desvio = 0F;
        int aux = 0;

        // mediana
        aux = list.size();
        if (aux % 2 == 0) {
            aux = (int) aux/2;
            mediana = (list.get(aux - 1) + list.get(aux))/2;
        } else {
            aux += 1;
            aux = (int) aux/2;
            mediana = list.get(aux - 1);
        }

        // variancia
        for (int num : list) {
            variancia += ((num - media) * (num - media));
        }
        if (list.size() - 1 != 0)
            variancia /= (list.size() - 1);
        else
            variancia /= list.size();

        // desvio padrão
        desvio = (float) Math.sqrt(variancia);

        System.out.printf("\n==> A mediana dos numeros digitados foi: %.2f", mediana);
        System.out.printf("\n==> A variancia dos numeros digitados foi: %.2f", variancia);
        System.out.printf("\n==> O desvio padrao dos numeros digitados foi: %.2f", desvio);
    }
}

File: Arrochado.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

#define MAX 100

int main(void)
{

    int maior = 0, menor = 0, size = 0, extremidade_menor = 0;
    int extremidade_maior = 100;
    char firstName[20];
    int tries[MAX];
    int input;
    int aux = 0;
    double media = 0.0;
    double mediana = 0.0;
    double variancia = 0.0;
    double desvio = 0.0;

    // gerando um numero randomicamente
    int randomNumber = rand() % 100;

    // nome do usuario
    printf("\nDigite seu nome:");
    gets(firstName);
    
    //jogo do arrochado
    do {
        printf("\n\nDigite seu palpite!\nDICA: O numero esta no intervalo de %d ~ %d:", extremidade_menor, extremidade_maior);
        scanf("%d", &input);

        // validar se esta no range definido
    	if (input > extremidade_maior || input < extremidade_menor)    
            printf("\n=>O numero digitado nao se encontra no range determinado de %d ~ %d\n  Tente novamente.", extremidade_menor, extremidade_maior);
        else {
            
            if (size == 0)
                menor = input;

            // armazena o maior e menor valor
            if (input > maior)
                maior = input;
            else if (input < menor)
                menor = input;

            if (input == randomNumber){
                tries[size] = input;
                size++;
                break;
            }

            if (randomNumber > input)
                extremidade_menor = input;
            else if (randomNumber < input)
                extremidade_maior = input;
            
            tries[size] = input;

            printf("=>Valor errado, tente novamente!\n");
            size++;
        };

    }while(input != randomNumber);

    printf("\n=>Parabens %s,\n  Voce acertou o numero!\n\n", firstName);

    // ordenar
    for (int i = 0; i < size; i++)
    {
        for (int j = 0; j < size; j++)
        {
            if(tries[i] < tries[j]){
                aux = tries[i];
                tries[i] = tries[j];
                tries[j] = aux;
            }
        }
    }

    for (int k = 0; k < size; k++)
    {
        media += tries[k];
        printf("[%d]", tries[k]);
    }
    
    // media
    media /= size;
    printf("\n==> A media dos numeros digitados foi %.2f", media);

    // mediana
    aux = size;
    if(aux % 2 == 0) {
        aux = (int) aux/2;
        mediana = (tries[aux - 1] + tries[aux])/2;
    } else {
        aux += 1;
        aux = (int) aux/2;
        mediana = tries[aux-1];
    }

    // variancia
    for (int z = 0; z < size; z++)
    {
        variancia += ((tries[z] - media) * (tries[z] - media));
    }

    if (size-1 != 0)
        variancia /= size - 1;
    else
        variancia /= size;
    
    // desvio
    desvio = sqrt(variancia);

    printf("\n==> O maior numero digitado foi %d e o menor %d", maior, menor);
    printf("\n==> A mediana dos numeros digitados foi %.2f", mediana);
    printf("\n==> A variancia dos numeros digitados foi %.2f", variancia);
    printf("\n==> O desvio padrao dos numeros digitados foi %.2f", desvio);

    return 0;
}

